src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployAllTask.java

```diff
@@ -45,8 +46,7 @@ public void setStageDirectory(File stageDirectory) {
   /** Task Entrypoint : Deploys the app and all of its config files. */
   @TaskAction
   public void deployAllAction() throws AppEngineException {
-    DeployExtension deployCopy = new DeployExtension(deployConfig);
-    deployCopy.getDeployables().clear();
+    DeployExtension deployCopy = new DeployExtension(deployConfig, new ArrayList<>());
```

Comment:
I think this might be a little bit of a strange pattern.

Would it be possible to just populate a local variable list with the files we know about and then call the constructor with that fully populated list?

Reply:
It needs the staging directory to resolve the path, so I don't think so. Unless I'm misunderstanding.

Reply:
oh, I mean you could do this copy further down in the deployAllAction method. First just create a local list of deployables, and then pass that one to new DeployExtension(deployConfig, deplyables)

Reply:
Oh I see what you mean.